src/common/low_precision_transformations/include/low_precision/network_helper.hpp

```diff
@@ -326,8 +326,8 @@ template <typename T, typename... Args>
 std::shared_ptr<Node> fold_reshape(Args&&... args) {
     std::shared_ptr<Node> node = std::make_shared<T>(args...);
     if (node->get_output_size() == 1) {
-        const auto data_const = ov::as_type_ptr<ov::opset1::Constant>(node->get_input_node_shared_ptr(0));
```

Comment:
@itikhono should we deprecated `get_input_node_shared_ptr` then? as unsafe method

Reply:
I think so
We observed many cases where this method leads to implicit issues.
e.g. 
https://github.com/openvinotoolkit/openvino/pull/29780
https://github.com/openvinotoolkit/openvino/pull/29659 
https://github.com/openvinotoolkit/openvino/pull/29976 

@praasz what do you think?

Reply:
As I understand the issue is for implicit conversion from Node shared pointer to output as there (when Node has many outputs). The `get_input_node_shared_ptr` member itself is not unsafe and is also used in other cases.
To remove it is better check other case like here. 

For `as_type_ptr` it should be fine but maybe here the raw pointer should be used instead.

The problem can be in cases like:
```cpp
       auto convert = std::make_shared<ov::snippets::op::ConvertSaturation>(
            result->get_input_node_shared_ptr(0),
            expected_type);
```
 when node is created from other Nodes pointers (default output) instead outputs.

Consider review this PR again if all changes are required.